contracts/registries/PoolRegistry.sol

```diff
@@ -0,0 +1,556 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import "../helper/BaseBoringBatchable.sol";
+import "../interfaces/ISwap.sol";
+import "../interfaces/ISwapGuarded.sol";
+import "../interfaces/IMetaSwap.sol";
+import "../interfaces/IPoolRegistry.sol";
+import "../meta/MetaSwapDeposit.sol";
+
+/**
+ * @title PoolRegistry
+ * @notice This contract holds list of pools deployed.
+ */
+contract PoolRegistry is
+    AccessControl,
+    ReentrancyGuard,
+    BaseBoringBatchable,
+    IPoolRegistry
+{
+    using SafeMath for uint256;
+
+    /// @notice Role responsible for managing pools.
+    bytes32 public constant SADDLE_MANAGER_ROLE =
+        keccak256("SADDLE_MANAGER_ROLE");
+    /// @notice Role responsible for managing community pools
+    bytes32 public constant COMMUNITY_MANAGER_ROLE =
+        keccak256("COMMUNITY_MANAGER_ROLE");
+    /// @notice Role that represents approved owners of pools.
+    /// owner of each pool must have this role if the pool is to be approved.
+    bytes32 public constant SADDLE_APPROVED_POOL_OWNER_ROLE =
+        keccak256("SADDLE_APPROVED_POOL_OWNER_ROLE");
+
+    /// @inheritdoc IPoolRegistry
+    mapping(address => uint256) public override poolsIndexOfPlusOne;
```

Comment:
Yeah that would be more explicit. I wont make changes here since it will change the ABIs as well. But added comments to these two variables to explain it better.